### DISCUSSION TITLE
Visualizer UI lib

### DISCUSSION THREAD

**Discussion by tinsaeDev** [dodona-edu/dolos#1464] [Q&A]
how to visualize the data received from analyzer ?
`const report = dolos.analyzePaths(["./file1.js", "./file2.js"]);`

**Comment by rien** [comment]
Hi @tinsaeDev, apologies for the late response.

Visualizing takes two steps:
- Writing the report data to CSV files
- Hosting the Web UI and CSV files, and opening this page with a browser

This functionality is currently implemented in `dolos-cli`, you can take a look here:

https://github.com/dodona-edu/dolos/blob/dc96168872506da37b4a3323df027c9d5f06fbb2/cli/src/cli/commands/run.ts#L187-L220

The `WebView` responsible for hosting the Web UI, actually depends on the `FileView` to write the CSV-files.

However, these classes are present in the package `@dodona/dolos`  where we don't export these classes, so you currently cannot do the visualization programmatically in your own code by importing these classes..

What is your use case for this? I could make these packages available or even provide some useful functions that would make this possible (for example a function `report.visualize()` that starts the Web UI).
